tests20/python_client/testcases/test_collection.py

```diff
@@ -2,11 +2,100 @@
 from base.client_request import ApiReq
 from utils.util_log import test_log as log
 from common.common_type import *
+from common.common_func import *
 
+default_schema = gen_default_collection_schema()
 
-class TestCollection(ApiReq):
+
+class TestCollectionParams(ApiReq):
     """ Test case of collection interface """
 
+    def teardown_method(self):
+        if self.collection.collection is not None:
+            self.collection.drop()
+
+    @pytest.fixture(
+        scope="function",
+        params=get_invalid_strs
+    )
+    def get_invalid_string(self, request):
+        yield request.param
+
+    # #5224
     @pytest.mark.tags(CaseLabel.L3)
-    def test_case(self):
-        log.info("Test case of collection interface")
+    def test_collection(self):
+        self._connect()
+        c_name = get_unique_str
+        log.info(c_name)
+        collection, _ = self.collection.collection_init(c_name, data=None, schema=default_schema)
+        assert collection.name == c_name
+        assert collection.description == default_schema.description
+        assert collection.schema == default_schema
+        assert collection.is_empty
+        assert collection.num_entities == 0
+        assert collection.primary_field is None
+        assert c_name in self.utility.list_collections()
+
+    def test_collection_empty_name(self):
+        """
+        target: test collection with empty name
+        method: create collection with a empty name
+        expected: raise exception
+        """
+        self._connect()
+        c_name = ""
+        ex, check = self.collection.collection_init(c_name, schema=default_schema)
+        assert "value  is illegal" in str(ex)
+
+    def test_collection_invalid_name(self, get_invalid_string):
+        """
+        target: test collection with invalid name
+        method: create collection with invalid name
+        expected: raise exception
+        """
+        self._connect()
+        c_name = get_invalid_string
+        ex, check = self.collection.collection_init(c_name, schema=default_schema)
+        assert "invalid" or "illegal" in str(ex)
+
+    # #5231
+    def test_collection_dup_name(self):
+        """
+        target: test collection with dup name
+        method: create collection with dup name and none schema and data
+        expected: collection properties consistent
+        """
+        self._connect()
+        c_name = get_unique_str
+        collection, _ = self.collection.collection_init(c_name, data=None, schema=default_schema)
+        assert collection.name == c_name
+        dup_name = c_name
+        dup_collection, _ = self.collection.collection_init(dup_name)
+        assert dup_name, c_name in self.utility.list_collections()
+        assert collection.name == dup_collection.name
+        log.debug(collection.schema)
```

Comment:
I think we need to assert the same collection is returned in 2nd init, collection.id == dup_collection.id ?
Also I think we can just use c_name to init twice, instead of making a dup_name

Reply:
1. no collection.id
2. I will delete dup_name

Reply:
what about id(collection) == id(dup_collection)